### PR TITLE
Remove #warning in the unnecessary places

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- libm/arm: Remove the compiler specific #warning
- Remove #warning if the code already return the error code or value
- net: Remove some minor #warning since it compiler specific

## Impact

Compiler time warning
Note: the error still report at the runtime

## Testing

CI

### CODE REVIEW

File: drivers/net/dm90x0.c [hunk 1]
```diff
@@ -1554,8 +1554,7 @@ static int dm9x_addmac(FAR struct net_driver_s *dev, FAR const uint8_t *mac)
 
   /* Add the MAC address to the hardware multicast routing table */
 
-#warning "Multicast MAC support not implemented"
-  return OK;
+  return -ENOSYS;
```
Comment (btashton) [inline drivers/net/dm90x0.c:1557]: For these missing logic sections I'm fine with the change to put the correct return code (that seems more correct) but should we also have a comment near it indicating missing logic? That would also make it easier to find and implement these sections later?

Comment (xiaoxiang781216) [reply drivers/net/dm90x0.c:1559]: Done, change #warning to comment for reference.

**Review by patacongo — APPROVED** [review]
It is hard to see what your long term plan is here for dealing with the `#warning` issue.  I would think that all #warnings should be handled in the same systematic way.  When that systematic way is implemented, will not these changes be unnecessary.

Is it acceptable to return -ENOSYS in these APIs?  Standard function descriptions enumerate the return values that they may produce.  Implying that other error codes should not be returned(?).  I tend to think that -ENOSYS or -ENOTSUP are probably OK if returned from any function.

-ENOSYS is the standard way of indicating that an entire function is not implemented.

> https://www.gnu.org/software/libc/manual/html_node/Error-Codes.html
>
> Macro: int ENOSYS
> 
>     “Function not implemented.” This indicates that the function called is not implemented at all, either in the C library itself or in the operating system. When you get this error, you can be sure that this particular function will always fail with ENOSYS unless you install a new version of the C library or the operating system. 

But a different error code should be returned if the function is implemented, but some specific feature is not supported:

> Macro: int ENOTSUP
> 
>     “Not supported.” A function returns this error when certain parameter values are valid, but the functionality they request is not available. This can mean that the function does not implement a particular command or option value or flag bit at all. For functions that operate on some object given in a parameter, such as a file descriptor or a port, it might instead mean that only that specific object (file descriptor, port, etc.) is unable to support the other parameters given; different file descriptors might support different ranges of parameter values.
> 
>     If the entire function is not available at all in the implementation, it returns ENOSYS instead. 

Shouldn't some of these be -ENOTSUP.

I am approving this now because the general change seems OK.  But we should at least consider some of these questions.

**Comment by jerpelea** [comment]
please add commit messages explaining the changes

**Comment by xiaoxiang781216** [comment]
But the commit title already explains the change clearly.

**Comment by davids5** [comment]
> But the commit title already explains the change clearly.

The "Why" is not in the messages. Only the "What" is. A lot of commit message state the "What" but never state the motivation (Why). 

The reason this is important is so that future contributors learn to not pattern match, but understand the motivations.

File: net/utils/net_chksum.c [hunk 2]
```diff
@@ -223,10 +223,6 @@ void net_chksum_adjust(FAR uint16_t *chksum,
                        FAR const uint16_t *optr, ssize_t olen,
                        FAR const uint16_t *nptr, ssize_t nlen)
 {
-#ifdef CONFIG_ENDIAN_BIG
-#  warning "Not verified on big-endian yet."
```
Comment (pkarashchenko) [inline net/utils/net_chksum.c:227]: Maybe also make it a comment?

Comment (xiaoxiang781216) [reply net/utils/net_chksum.c:227]: I review the code, it looks good.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**